Support for cosedimenting components

doluk opened the pull request.

ehb54 commented:
@doluk 
This seems ok to merge, waiting for an update on the [us3_sql](https://github.com/ehb54/us3_sql/pull/1) pull request.

Note - updating any system with database changes generally requires careful handling (either upgrading the db with a standard procedure or manually inserting the new tables and stored procedures)

doluk commented:
Should I write an update procedure for this too? Where should I put it if so? @ehb54

doluk commented:
In this case the update only consists in adding new tables and procedures. I don't know if the time is worth it this time. If you want one, I can do it (after I figured out what the good practice for such things is xD)

doluk commented:
@ehb54 
I added in the pull request an sql file containing all stuff I changed so it is just executing this file. Let me know if you want something changed (or change it yourself, if you want)
